### DISCUSSION TITLE
Vaildator-Link for local files without server/localhost

### DISCUSSION THREAD

**Discussion by buhtz** [validator/validator#1338] [Q&A] [answered]
From [hugo](https://kodify.net/hugo/tools/html-validator-online/#create-a-local-only-html-validator-link-in-hugo) I know validator links like this.
```
{{ if in (.Site.BaseURL | string) "localhost" }}
    <a href="https://validator.w3.org/nu/?doc={{ "https://www.example.com" }}{{ .RelPermalink }}"
    target="_blank">Validate @ W3C</a>
{{ end }}
```
But they only work if the content to validate is accessible via an URL on a web server.

I only have HTML files on my local filesystem. There is no server not even on localhost. I just double click on the html file and the browser comes up and shows me a `file:///` url in the url bar. Now I want to have a validate link at the bottom of such a page/file. Is this possible?

It is like the file-upload feature of yours.
https://validator.w3.org/nu/#file

**Comment by j9t** [comment, marked as answer]
So you’d like to do local validation as well, or pass the file source on to the public validator?

If you do want to go for local validation, how about setting up the validator locally? It requires some judo, but you can [use this validator to validate files on the command line](https://meiert.com/blog/html-validator-packages/), and even have it do batch validation.

I don’t think the validator allows for passing on the source directly (though on writing this, maybe there’s an option to POST).

**Reply by buhtz** [reply]
Hui, this question is quit old. As I remember it was in context of [Hyperorg](https://codeberg.org/buhtz/hyperorg). My intention was pass the (local) file source on to the public validator. Of course it would be much more preferable if there would be an local running (CLI) validator, too. Most of my users wouldn't want to expose the content of their local HTML files to a public validator.
